key.ui/src/main/java/org/key_project/util/java/SwingUtil.java

```diff
@@ -149,4 +150,46 @@ public static JScrollPane createScrollPane(JTable table) {
 
         return scrollPane;
     }
+
+    /**
+     * Try to find a tree path after updating the nodes of a {@link JTree}.
+     * This will compare nodes by their string representation.
+     *
+     * @param tree the tree to search
+     * @param tp the tree path
+     * @return a tree path or null
+     */
+    public static TreePath findPathInNewTree(JTree tree, TreePath tp) {
+        TreePath root = tree.getPathForRow(0);
+        Object last = root.getLastPathComponent();
+        int pathIndex = 0;
+        if (!last.toString().equals(tp.getPathComponent(pathIndex).toString())) {
```

Comment:
Why does this have to be the toString repr? Filtering shouldn't change the nodes so why is reference equality not enough?

Reply:
The nodes are cleared here: https://github.com/KeYProject/key/blob/0ac6d7f2f89621db5bcd223c76470fdcc857ac4d/key.ui/src/main/java/de/uka/ilkd/key/gui/prooftree/GUIProofTreeModel.java#L377-L380

Reply:
I think your approach would be more efficient. I'll use it instead if it works as expected.